src/com/itacademy/eshop/services/ShopService.java

```diff
@@ -25,7 +28,22 @@ private ArrayList<Product> createProducts() {
          * Each class must extend AbstractProduct class.
          * So you will have to create at least 6 classes that extend Product class.
          */
-        throw new UnsupportedOperationException("createProducts() method is not implemented yet.");
-    }
+        //create new array list
+        //Product product = new Product();// these are both objects
+        //contructor creates objects
+        //ArrayList electronics = new ArrayList<>();
+        //child and parent class fields
+        ArrayList<Product> electronics = new ArrayList<>();
+        ArrayList<Product> clothings = new ArrayList<>();
+
+        //took created array list and added new computer object
+        electronics.add(new Computer("Laptop", 100, Category.ELECTRONICS, "123", "Dell", "intel", 12));
 
+        clothings.add(new Clothing("SHirt", 100, Category.CLOTHING, "leather", "winter"));
```

Comment:
clothings is assigned and modified, but never used. in this case you should add all products to the electronics list